extensions/amp-inline-gallery/1.0/amp-inline-gallery.js

```diff
@@ -37,13 +39,59 @@ import {userAssert} from '../../../src/log';
 const TAG = 'amp-inline-gallery';
 
 class AmpInlineGallery extends PreactBaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.carousel_ = null;
+
+    /** @private {?ResizeObserver} */
+    this.resizeObserver_ = null;
+  }
+
   /** @override */
   init() {
+    this.carousel_ = this.element.parentElement.querySelector(
+      'amp-base-carousel'
+    );
+    this.resizeObserver_ = new this.win.ResizeObserver(() => {
+      // If carousel resizes, update height variable.
+      setStyle(
+        this.element,
+        '--i-amphtml-carousel-height',
+        px(this.carousel_.offsetHeight)
+      );
+    });
+    this.resizeObserver_.observe(this.carousel_);
+
     return dict({
       'children': <ContextExporter shimDomElement={this.element} />,
     });
   }
 
+  /** @override */
+  mutationObserverCallback(entries) {
+    if (this.carousel_ && this.carousel_.parentElement !== this.element) {
+      // Originally observed element has been detached.
+      this.resizeObserver_.unobserve(this.carousel_);
+      this.carousel_ = this.element.parentElement.querySelector(
+        'amp-base-carousel'
+      );
+      if (this.carousel_) {
+        this.resizeObserver_.observe(this.carousel_);
+      }
+    }
+    if (entries.every((entry) => entry.attributeName === 'style')) {
+      return;
+    }
+    setStyle(
+      this.element,
+      '--i-amphtml-carousel-top',
```

Comment:
This is still very fragile. There many possible reasons why the carousel's `offsetTop` can change. Or even the interpretation of the `offsetTop` value itself is pretty questionable. We'd need to crawl the chain of `offsetParent` between the carousel and gallery to actually get the right value.

Reply:
I know, it is really not ideal. I wonder what is a better way to measure and adapt, or if we are just forcing something that is not the best way to implement. I'll think about some other approaches.